### ISSUE TITLE
CenterAnimation configuration

### ISSUE THREAD

**Issue by agerstown** [SwiftKickMobile/SwiftMessages#134]
Hello!

I would like to use CenterAnimation, but it can be shown in the center of the screen only. Could you please allow to configure position of the notification on the screen? For example, I need to show the notification with the offset 100pt from top using the same animation used in CenterAnimation, but it is only possible to show it in the center.

I tried to replicate CenterAnimation, but it is not possible as it uses panHandler (PhysicsPanHandler) which has an internal level. Can't subclass CenterAnimation either as it doesn't have an open level.

**Comment by wtmoose** [comment]
Hi. Absolutely! I realized that the name `CenterAnimation` was a mistake shortly after releasing it 😀

**Comment by wtmoose** [comment]
Could you try the `work/physicsAnimationBranch` and let me know if if works for you?

````swift
let animator = PhysicsAnimation()
animator.placement = .top
config.presentationStyle = .custom(animator: animator)
````

To have the offset be 100pt from the top, you'd need to modify the constraints in the nib file. You can do this by copying `CenterView.nib` into your project (assuming you're using that nib file) and making the following modifications to the top constraint on `Background View`:

1. Turn off "Relative to Margin"
2. Set the constant to 100

I also made `PhysicsPanHandler's`, so you should be able to use that in a custom animation if you want.

**Comment by agerstown** [comment]
Hello! Sorry for the late reply.

Just checked this new branch and it works! Thank you so much.

One problem: as I set the margin to a background view of the alert view, other views under that view can't be tapped as they are overlapped. I use the custom view, could that be a problem? I attached the screenshot which should explain the problem. 

![screen shot 2017-11-09 at 12 51 44](https://user-images.githubusercontent.com/13255497/32600138-ad129174-c54f-11e7-9a54-38c391cc383d.png)

I managed to get the result I need using custom animation (modified PhysicsAnimation): 

I created custom init with a `topOffset` parameter, then in the `func install(context: AnimationContext)` I changed top constant to the one I need:
`NSLayoutConstraint(item: view, attribute: .top, relatedBy: .equal, toItem: container, attribute: .top, multiplier: 1, constant: topOffset).isActive = true`

**Comment by agerstown** [comment]
I also found one other problem (or may be it is supposed to work like that):
sometimes when you slightly move an alert it doesn't go away from the screen, but is placed in the center of it:
![2017-11-09 13 24 47](https://user-images.githubusercontent.com/13255497/32600716-8fb50d6c-c551-11e7-9eeb-f28db7cacf3c.jpg)

**Comment by wtmoose** [comment]
Hi. I fixed the issue with snapping back to center.

Regarding your issue with elements behind the message view not being tappable, it looks like you just need to set the `backgroundView` property to the white view. Otherwise, it will default to the message view itself and the entire message view will intercept taps (assuming you're using `MessageView`).

I've attached a sample project to demonstrate. The steps I used were:

1. Copy CardView.xib into my project and add it to my app target.
2. Modify the top constraint between the `backgroundView` and the superview's margin from 10 to 50.

Hope this helps.

[SwiftMessagesTest.zip](https://github.com/SwiftKickMobile/SwiftMessages/files/1458014/SwiftMessagesTest.zip)
